File: JH/DAY12/10845.cpp
#include <bits/stdc++.h>
using namespace std;

queue<int>q;

int main(){


    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);


    int n;
    cin >> n;

    for (int i = 0; i <n ; ++i) {
        string s;
        cin >> s;

        if(s=="push"){
            int x;
            cin >> x;
            q.push(x);
        }
        if(s=="pop"){
            if(q.size()==0){
                cout << "-1" << "\n";
            }else{
                cout << q.front() <<"\n";
                q.pop();
            }

        }
        if(s=="size"){
            cout << q.size()<<"\n";
        }
        if(s=="empty"){
            if(q.size()==0){
                cout << "1" <<"\n";
            }else{
                cout << "0" << "\n";
            }
        }
        if(s=="front"){
            if(q.size()==0){
                cout << "-1" << "\n";
            }else{
                cout << q.front() << "\n";
            }
        }
        if(s=="back"){
            if(q.size()==0){
                cout << "-1" << "\n";
            }else{
                cout << q.back() << "\n";
            }
        }

    }




}

File: SY/queue/10845.py
import sys

n=int(sys.stdin.readline())
li=[sys.stdin.readline().strip('\n') for _ in range(n)]
que=[]
for i in li:
	if 'push' in i:
		que.append(i[5:])
	elif i=='pop':
		if que:
			print(que.pop(0))
		else:
			print(-1)
	elif i=='size':
		print(len(que))
	elif i=='empty':
		if not que:
			print(1)
		else:
			print(0)
	elif i=='front':
		if que:
			print(que[0])
		else:
			print(-1)
	elif i=='back':
		if que:
			print(que[-1])
		else:
			print(-1)